servers/rendering/renderer_viewport.cpp

```diff
@@ -843,7 +843,7 @@ void RendererViewport::draw_viewports(bool p_swap_buffers) {
 				// commit our eyes
 				Vector<BlitToScreen> blits = xr_interface->post_draw_viewport(vp->render_target, vp->viewport_to_screen_rect);
 				if (vp->viewport_to_screen != DisplayServer::INVALID_WINDOW_ID) {
-					if (OS::get_singleton()->get_current_rendering_driver_name().begins_with("opengl3")) {
+					if (RSG::rasterizer->is_opengl()) {
```

Comment:
Ultimately, the compositor determines if it requires `gl_end_frame` be called, so it can determine that by returning `true` for `is_opengl`.